File: Contests/Yandex-Contest_IWO-2023/python/1-crowd-scenes.py
def min_people_days(n, m, scenes):
    scenes.sort(key=lambda x: x[0])  # Сортируем сцены по дням начала съемок
    hired_actors = [0] * n  # Список для хранения количества актеров, нанятых в каждый день

    for scene in scenes:
        l, r, x = scene
        for day in range(l-1, r):
            if hired_actors[day] < x:
                hired_actors[day] = x

    total_people_days = sum(hired_actors)

    return total_people_days

# Чтение входных данных
n, m = map(int, input().split())
scenes = [list(map(int, input().split())) for _ in range(m)]

# Вызов функции и вывод результата
result = min_people_days(n, m, scenes)
print(result)

File: Contests/Yandex-Contest_IWO-2023/1-crowd-scenes.swift
func minPeopleDays(n: Int, m: Int, scenes: [[Int]]) -> Int {
    var scenes = scenes
    scenes.sort { $0[0] < $1[0] } 
    var hiredActors = Array(repeating: 0, count: n)

    for scene in scenes {
        let (l, r, x) = (scene[0], scene[1], scene[2])
        for day in (l-1)..<r {
            if hiredActors[day] < x {
                hiredActors[day] = x
            }
        }
    }

    let totalPeopleDays = hiredActors.reduce(0, +)

    return totalPeopleDays
}

let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0]
let m = input[1]

var scenes = [[Int]]()
for _ in 0..<m {
    let sceneInput = readLine()!.split(separator: " ").map { Int($0)! }
    scenes.append(sceneInput)
}

let result = minPeopleDays(n: n, m: m, scenes: scenes)
print(result)
